### PR TITLE
Fix generation of alias when using FQCN with TableLocator::get().

### LINKED ISSUE #14948: Sqlite error when no table alias set and class named used with locator->get()
This is a (multiple allowed):

* [x] bug
* [x] enhancement

* CakePHP Version: 4.x

### What you did

```php
class FileStorageTestTable extends Table
{
    public function initialize(array $config): void
    {
        parent::initialize($config);
        $this->setTable('file_storage');
    }
}

$this->FileStorage = $this->getTableLocator()->get(FileStorageTestTable::class);
$this->FileStorage->get('uuid');
```

### What happened

```
1) Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageBehaviorTest::testAfterDelete
PDOException: SQLSTATE[HY000]: General error: 1 unrecognized token: "\"
```

The reason is that the class name is used.

```sh
########## DEBUG ##########
'SELECT Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable.id AS Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable__id, Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable.user_id AS Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable__user_id, Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable.foreign_key AS Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable__foreign_key, Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable.model AS Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable__model, Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable.filename AS Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable__filename, Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable.filesize AS Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable__filesize, Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable.mime_type AS Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable__mime_type, Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable.extension AS Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable__extension, Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable.hash AS Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable__hash, Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable.path AS Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable__path, Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable.adapter AS Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable__adapter, Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable.variants AS Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable__variants, Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable.metadata AS Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable__metadata, Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable.created AS Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable__created, Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable.modified AS Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable__modified FROM file_storage Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable WHERE Burzum\FileStorage\Test\TestCase\Model\Behavior\FileStorageTestTable.id = :c0 LIMIT 1'
```

### What you expected to happen

Simply no error or a **meaningful** exception message.

I'm not sure but maybe it is possible to detect this case somehow by checking if alias is set and a class name is used something like  this pseudo code `if alias === null && class_exists($name)` then either set an alias based on something or throw at least an exception that tells me about the cause and tells me to set an alias.

### PR DESCRIPTION
**Pull request by ADmad**
Closes #14948

### CODE REVIEW

**Review by dereuromark — APPROVED** [review]

File: src/ORM/Locator/TableLocator.php [hunk 1]
```diff
@@ -160,8 +160,14 @@ public function get(string $alias, array $options = []): Table
...
     {
-        [, $classAlias] = pluginSplit($alias);
-        $options = ['alias' => $classAlias] + $options;
+        if (strpos($alias, '\\') === false) {
+            [, $classAlias] = pluginSplit($alias);
+            $options = ['alias' => $classAlias] + $options;
+        } elseif (!isset($options['alias'])) {
+            $options['className'] = $alias;
+            /** @psalm-suppress PossiblyFalseOperand */
+            $alias = substr($alias, strrpos($alias, '\\') + 1, -5);
```
Comment (othercorey) [inline src/ORM/Locator/TableLocator.php:169]: What does this look like when you give it an invalid table class? Can we add a test for that?

Comment (markstory) [reply src/ORM/Locator/TableLocator.php:169]: Looks like the last 5 characters will get chopped off.

Comment (dereuromark) [reply src/ORM/Locator/TableLocator.php:169]: It wouldnt be worse than the current code, for sure.
But we can also check and throw a meaningful exception in that case.

Comment (ADmad) [reply src/ORM/Locator/TableLocator.php:169]: I copied this code from `ModelAwareTrait::loadModel()`. 

For an invalid class name the method will return a `Table` instance with whatever alias is generated after this string manipulation. I don't think any extra defensive coding is necessary here.

Comment (othercorey) [reply src/ORM/Locator/TableLocator.php:169]: Sounds good. I don't want extra coding either ... was just curious what error the user would see.

**Pull request merged**